sdk/python/kfp/dsl/_types.py

```diff
@@ -0,0 +1,166 @@
+		"type": "string",
+		"pattern": "^gs://.*$"
+	}
+
+	def __init__(self, path_type='', file_type=''):
+		'''
+		Args
+		:param path_type: describes the paths, for example, bucket, directory, file, etc
+		:param file_type: describes the files, for example, JSON, CSV, etc.
+		'''
+		self.path_type = path_type
+		self.file_type = file_type
+
+class GCRPath(BaseType):
+	openapi_schema_validator = {
+		"type": "string",
+		"pattern": "^.*gcr\\.io/.*$"
+	}
+
+class GCPRegion(BaseType):
+	openapi_schema_validator = {
+		"type": "string"
+	}
+
+class GCPProjectID(BaseType):
+	'''MetaGCPProjectID: GCP project id'''
+	openapi_schema_validator = {
+		"type": "string"
+	}
+
+# General Types
+class LocalPath(BaseType):
+	#TODO: add restriction to path
+	openapi_schema_validator = {
+		"type": "string"
+	}
+
+class InconsistentTypeException(Exception):
+	'''InconsistencyTypeException is raised when two types are not consistent'''
+	pass
+
+def check_types(checked_type, expected_type):
+	'''check_types checks the type consistency.
+	For each of the attribute in checked_type, there is the same attribute in expected_type with the same value.
+	However, expected_type could contain more attributes that checked_type does not contain.
+	Args:
+		checked_type (BaseType/str/dict): it describes a type from the upstream component output
+		expected_type (BaseType/str/dict): it describes a type from the downstream component input
+		'''
+	if isinstance(checked_type, BaseType):
+		checked_type = _instance_to_dict(checked_type)
+	elif isinstance(checked_type, str):
+		checked_type = _str_to_dict(checked_type)
+	if isinstance(expected_type, BaseType):
+		expected_type = _instance_to_dict(expected_type)
+	elif isinstance(expected_type, str):
+		expected_type = _str_to_dict(expected_type)
+	return _check_dict_types(checked_type, expected_type)
+
+def _check_valid_type_dict(payload):
+	'''_check_valid_type_dict checks whether a dict is a correct serialization of a type
+	Args:
+		payload(dict)
+	'''
+	if not isinstance(payload, dict) or len(payload) != 1:
+		return False
+	for type_name in payload:
+		if not isinstance(payload[type_name], dict):
+			return False
+		property_types = (int, str, float, bool)
+		for property_name in payload[type_name]:
+			if not isinstance(property_name, property_types) or not isinstance(payload[type_name][property_name], property_types):
+				return False
+	return True
+
+def _instance_to_dict(instance):
+	'''_instance_to_dict serializes the type instance into a python dictionary
+	Args:
+		instance(BaseType): An instance that describes a type
+
+	Return:
+		dict
+	'''
+	return {type(instance).__name__: instance.__dict__}
+
+def _str_to_dict(payload):
+	import json
+	json_dict = json.loads(payload)
+	if not _check_valid_type_dict(json_dict):
+		raise ValueError(payload + ' is not a valid type string')
+	return json_dict
+
+def _check_dict_types(checked_type, expected_type):
+	'''_check_type_types checks the type consistency.
```

Comment:
nit: _check_dic_types

Reply:
good catch. I have updated the comments in the following PRs. Will merge this one first.